spec/Section 3 -- Type System.md

```diff
@@ -1147,6 +1162,8 @@ Interface types have the potential to be invalid if incorrectly defined.
 5. If an interface type declares that it implements one or more interfaces that
    implement other interfaces, it must also include each transitively
    implemented interface in its list of implemented interfaces.
+6. An interface cannot implement an interface that in turn implements the
+   original interface.
```

Comment:
Since we decided that interfaces should implement all transitive interfaces, this means we can simplify this rule to `An interface cannot implement itself`.

Reply:
This is a great point; I've updated the language accordingly. One additional benefit of this wording is the symmetry with our implementation in graphql-js.